streaming/base/stream.py

```diff
@@ -138,13 +140,54 @@ def __init__(self,
                 raise ValueError('`download_timeout` must be positive')
             self.download_timeout = download_timeout
 
+        self.local_directory_timeout = local_directory_timeout
+        if local_directory_timeout <= 0:
+            raise ValueError('`local_directory_timeout` must be positive')
+
         self.validate_hash = validate_hash
 
+        if local is None:
+            self.local = self._get_temporary_directory()
+            self._create_or_wait_for_local()
+        else:
+            self.local = local
+
         self._keep_zip = keep_zip
         if keep_zip is not None:
             self.keep_zip = keep_zip
             self.safe_keep_zip = self.keep_zip or self.remote in {None, self.local}
 
+    def _get_temporary_directory(self) -> str:
+        """Construct a path to a temporary directory based on remote and split."""
+        root = tempfile.gettempdir()
+        hash = None
+        if self.remote is not None:
+            hash = hashlib.md5(self.remote.encode('utf-8')).hexdigest()
+        # Removes underscore if self.split is an empty string
+        folder = '_'.join(filter(None, [hash, self.split]))
```

Comment:
need it to be like

```
root/
    hash/
        split/
```

so just `return os.path.join(root, hash, split)` i presume